Sources/AnyCodable/AnyEncodable.swift

```diff
@@ -122,36 +103,34 @@ extension _AnyEncodable {
         }
     }
 
-    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+    #if canImport(Foundation)
     private func encode(nsnumber: NSNumber, into container: inout SingleValueEncodingContainer) throws {
-        switch CFNumberGetType(nsnumber) {
-        case .charType:
+        switch Character(Unicode.Scalar(UInt8(nsnumber.objCType.pointee)))  {
```

Comment:
I had change to use _[NSNumber.objCType](https://developer.apple.com/documentation/foundation/nsvalue/1412365-objctype/)_ instead of _[CFNumberGetType(_:)](https://developer.apple.com/documentation/corefoundation/1543388-cfnumbergettype/)_ which requires toll-free bridging between _NSNumber_ and _CFNumber_ since, Swift 5.1.5 in Linux does not support this behaviour.

Reply:
Oh, interesting. Character codes are a downgrade from a real enumeration type, but if this works, then I'm all for it.